### PR TITLE
Link to docs.twisted.org

### PR DESCRIPTION
**Pull request by twm**
Update the location of the Twisted API docs, which should fix the linkchecker.

See also https://github.com/twisted/treq/pull/351

### CODE REVIEW

**Comment by twm** [comment]
The 404 at https://docs.twisted.org/en/stable/api/twisted.web.template.Element.html appears to be a Pydoctor issue.

However the [`Element` symbol is exported by this module](https://github.com/twisted/twisted/blob/26c3c04e55d0b25532709e1e1dc5d1433dde80f3/src/twisted/web/template.py#L25).

I see that this has been fixed in the Pydoctor used for trunk: https://docs.twisted.org/en/latest/api/twisted.web.template.Element.html works.

I guess I'll just wait for a Twisted release.

**Comment by adiroiban** [comment]
I have reverted my last commit as it was a bad one.

It would be nice to have the .py examples as external files, to make it easier to manually test the examples... or even write automated  tests for them :)

--------

I will look to make a new Twisted release so that we fix the docs.

Thanks!

File: docs/_extensions/apilinks.py [hunk 1]
```diff
@@ -13,6 +13,8 @@
 from types import MappingProxyType
 
 
+apilinks_base_url = "https://docs.twisted.org/en/stable/api/"
```
Comment (adiroiban) [inline docs/_extensions/apilinks.py:16]: Maybe use latest twisted docs for now

```suggestion
apilinks_base_url = "https://docs.twisted.org/en/latest/api/"
```

----------

I see `:api:` custom directive is used 38 times.
Maybe after twitsted 22.8.0 is relased, in a separate PR we (I) can migrate to intrsphinx links and remove the custom extension.

In twisted/twisted we no longer use the `:api:` custom directive and I have submitted a PR to remove that code.

**Review by adiroiban — APPROVED** [review]
Looks good.

My suggestion is to go with "latest" docs for api links so that we can merge this and also merge the context manager PR.

I have started the release process for twisted, I hope it will be ready by the start of september , but I think is best to not block this PR.

Thanks again

**Comment by glyph** [comment]
@adiroiban looks like a bunch of merge conflicts have crept in here, could you have another look at this?

**Comment by glyph** [comment]
oops, I guess @twm is the author here and @adiroiban is the reviewer, so cc: @twm for conflict resolution

**Comment by adiroiban** [comment]
I forgot about this PR.

I think that many of the changes here, I have "reinvented" them in the PR that is not merged https://github.com/twisted/klein/pull/573

I think that .gitignore is the only thing left.

----------

btw, it would be awesome to build the docs outside of the docs source folder... makes it easier to do quick grep/search for some word/type without having the exclude all the build files

File: docs/_extensions/apilinks.py (hunk 1 above)
Comment (twm) [reply docs/_extensions/apilinks.py:16]: Seems to work now!

**Comment by twm** [comment]
@adiroiban Thanks for the review, and @glyph thanks for the poke!

**Pull request merged**